### ISSUE TITLE
Is the randomly generated hash configurable?

### ISSUE THREAD

**Issue by kkebb** [phoronix-test-suite/phoronix-test-suite#135]
Hello again,

I noticed that the hash CODE gets randomly generated every time we start up a new phoromatic server, is there a way to configure that CODE so that it's consistent every time we bring up the server? This would be really helpful in the use case of orchestration with docker where the same code can be used by the clients when they need to connect to the server and helps avoid writing a script to extract the code from the server and push it to the clients.

**Comment by michaellarabel** [comment]
Can you clarify what "hash CODE" you are talking about? The Phoromatic clients should only need to know the Phoromatic server's IP address and port number along with the account identifier. Assuming you are using the same account and same port (port is configurable via the XML configuration file) and that the server IP isn't changing, not sure what changing hash code you are talking about.

**Comment by kkebb** [comment]
My apologies, the "hash code" I am referring to is the "account identifier" that you mentioned, I just used different terminologies lol

**Comment by michaellarabel** [comment]
This account identifier should never be changing, unless you are creating a new account each time you start your docker phoromatic-server instance (in which case, sounds like your /var/lib/phoronix-test-suite/phoromatic/ or ~/.phoronix-test-suite/phoromatic/ (non-root) is getting hosed). Can you clarify? Not aware of any bugs at all nor any code paths where that account identifier would be randomly changing.

**Comment by kkebb** [comment]
Here is an example of what I am referring to:

    localhost:8000/9QZJ8Y

I would like to make the _9QZJ8Y_ static, otherwise if that is not possible(i.e. configurable) then I will need a way to extract it using a script.

**Comment by michaellarabel** [comment]
That ID (9QZJ8Y in your reported case) should be static. It shouldn't be changing, that's what I am saying. If it's changing for you, are you like making a new account on the Phoromatic Server each time or what is the workflow you are doing on launching the Phoromatic Server.

**Comment by kkebb** [comment]
Right - so you are correct in assuming that I am making a new account each time because that is part of my workflow when I use the docker orchestration tool: "docker-compose".

In more detail, the workflow that I will be attempting to use is docker's orchestration tool "docker-compose". The orchestrator will be creating a phoromatic server container and running it first, and then I will be turning on the client containers thereafter, however, I will need to have the ID **before** I startup the clients in order to connect them to the right server. This is currently difficult to automate (i.e. using the docker-compose tool) since I don't know what the ID will be ahead of time. The reason why I need it to be "configurable" (i.e. static every time I bring up a new phoromatic server) is because I will be tearing the orchestration down and cleaning up the VMs from the phoronix suite containers, thus I will be running them again against the same, or different, systems.

I hope this clarifies my use-case..

**Comment by michaellarabel** [comment]
I don't quite understand why you don't use the same account rather than always making a new account. However, from the account creation page you can pre-seed your desired account ID using the ?seed_accountid=[the string] on that Phoromatic web page. If the seed_accountid GET variable is set, it will use that rather than coming up with its own random string. Does that work for your purposes?

**Comment by kkebb** [comment]
Is there a command line tool, where I can set up the ?seed_accountid variable prior to turning on the phoromatic server?

**Comment by kkebb** [comment]
I think I have found where I can edit it on the UI, so nevermind that question from my comment above. However, is there a way to configure that ?seed_accountid prior to/after starting the phoromatic server using phoronix-test-suite CLI?

**Comment by michaellarabel** [comment]
There isn't a way to do it from the CLI since at the moment there isn't such option to add a new Phoromatic account via the CLI. And setting that via like some global environment variable wouldn't work since if multiple people then try creating multiple accounts, they would hit a collision with the same account ID. If you are able to upload your script or so for however you are deploying it, I might have some other advice or thoughts.

**Comment by kkebb** [comment]
well currently I haven't written any code yet to deploy Phoronix - docker is taking care of the deployment - what would be nice to have is the account_id configurable (either through REST or CLI) that way it is a pre-determined value and the automation becomes simpler from a phoronix user perspective.

**Comment by michaellarabel** [comment]
Once you have some code written or at least a more thorough explanation of workflow / intended deployment strategy, let me know. For now it frankly doesn't make too much sense and in the Phoromatic Server deployments I'm aware of, no one has ever requested functionality along the lines of what you're desiring since once everyone makes their Phoromatic accounts, the account ID is static and shouldn't change for that account.

**Comment by kkebb** [comment]
So here is the deployment strategy (using maestro-ng orchestration for docker):

    __maestro:
      schema: 2

    name: SYSTEM_BENCHMARKING

    ship_defaults:
      docker_port: 2375
      timeout: 30

    ships:

      pts01: {ip: <HOST1>}
      pts02: {ip: <HOST2>}
      pts03: {ip: <HOST3>}
      pts04: {ip: <HOST4>}
    
    services:
      phoromatic_server:
         image: <HOST_URL>/docker-phoronix:b2
         instances:
            ps:
                ship: pts01
        ports: {client: "8000:8000", ws: "8001:8001"}
        lifecycle:
          running: [{type: tcp, port: ws}]
        command: "start-phoromatic-server"

      pts_cli:
        image: <HOST_URL>/docker-phoronix:b2
        requires: [ phoromatic_server ]
        instances:
          pts_zk1:
            ship: pts01
            command: "phoromatic.connect {{ env.PHOROMATIC_SERVER }}"
          pts_kafka1:
            ship: pts01
            command: "phoromatic.connect {{ env.PHOROMATIC_SERVER }}"
          pts_wf1:
            ship: pts01
            command: "phoromatic.connect {{ env.PHOROMATIC_SERVER }}"
          pts_zk2:
            ship: pts02
            command: "phoromatic.connect {{ env.PHOROMATIC_SERVER }}"
          pts_kafka2:
            ship: pts02
            command: "phoromatic.connect {{ env.PHOROMATIC_SERVER }}"
          pts_wf2:
            ship: pts02
            command: "phoromatic.connect {{ env.PHOROMATIC_SERVER }}"

The reason why we would like the hash be configurable is because it enables us to avoid having to inject it manually when we wish to run the PTS tool in an orchestrated manner.

A little break down of the above YAML file follows:
There are 2 _services_ defined above, `phoromatic_server` and `pts_cli`. The `phoromatic_server` service is depended on by the `pts_cli` service (note the `requires: [ phoromatic_server ]` line).

Each _service_ is comprised of 1-many **instances** (i.e. container(s)). As you can see, there is **only one** container for the `phoromatic_server` and **multiple** containers for the `pts_cli` service. Each of the `pts_cli` containers are given their respective `phoromatic.connect` command. My main issue at this point is in terms of orchestration of the Phoronix "mesh" (aka server-client relationship), where I am obliged to start up the server AND create an account before I can get a randomly generated hash, in which I can then go into the UI and copy and paste (not that they are very tedious steps) and inject them into the `env.PHOROMATIC_SERVER` environment variable in order for maestro-ng to pick them up and inject them into the `pts_cli` containers. It would be a lot simpler to enable account creation from the CLI where in turn we can have the hash either pushed into a file somewhere inside of the server container OR make Phoromatic expose an API call in which the clients can invoke using curl and inject it into their `phoromatic.connect` command. The third (easier) option would be to have the hash configurable in the same file where the ports are configured.

Hope that this clarifies why there is a need for the hash to be configurable/dynamically created and available for ease of automation and orchestration.

**Comment by michaellarabel** [comment]
Apologies for losing track of this thread. The Phoromatic account ID can be pre-seeded if setting it via seed_accountid as POST/GET request. E.g. if going to your Phoromatic server like: http://192.168.1.110:8888/?seed_accountid=5555 and the registration form on that same page is filled out, the subsequently created account will have been seeded to the defined value. It's actually been in Phoromatic for about two years now but had forgot about that unadvertised functionality. I hope that works out for you otherwise let me know.
